nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -376,6 +376,59 @@ in
           If enabled (the default), checks that Nix can parse the generated nix.conf.
         '';
       };
+
+      registry = mkOption {
+        type = types.attrsOf (types.submodule (
+          let
+            inputAttrs = types.attrsOf (types.oneOf [types.str types.int types.bool types.package]);
+          in
+          { config, name, ... }:
+          { options = {
+              from = mkOption {
+                type = inputAttrs;
+                example = { type = "indirect"; id = "nixpkgs"; };
+                description = "The flake reference to be rewritten.";
+              };
+              to = mkOption {
+                type = inputAttrs;
+                example = { type = "github"; owner = "my-org"; repo = "my-nixpkgs"; };
+                description = "The flake reference to which <option>from></option> is to be rewritten.";
+              };
+              flake = mkOption {
+                type = types.unspecified;
+                default = null;
+                example = literalExample "nixpkgs";
+                description = ''
+                  The flake input to which <option>from></option> is to be rewritten.
```

Comment:
It's not clear to me what "a flake input" means here. Input seems to be a role, rather than an intrinsic description of what it is that you need here. I guess you could equally say "a flake output attrset", because that's what defines an invoked flake. It'd also be helpful to know by which means this configuration is stored.
What happens when you use this option in a flake where you use `inputs.<input>.follows`? I don't think that important context preserved, is it?